catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/MetacardMarshallerImpl.java

```diff
@@ -117,11 +117,14 @@ public String marshal(Metacard metacard, Map<String, Serializable> arguments)
             throws XmlPullParserException, IOException, CatalogTransformerException {
         PrintWriter writer = this.writerProvider.build(Metacard.class);
 
-        Boolean omitXmlDec = (Boolean) arguments.get(OMIT_XML_DECL);
-        if (omitXmlDec == null || !omitXmlDec) {
-            writer.setRawValue(XML_DECL);
+        if (arguments != null && arguments.get(OMIT_XML_DECL) != null) {
+            Boolean omitXmlDec = Boolean.valueOf(String.valueOf(arguments.get(OMIT_XML_DECL)));
+            if (omitXmlDec == null || !omitXmlDec) {
```

Comment:
Having checked to ensure that there's a non-null value in the map in the outer condition, you can skip this null check here. Any non-null value that you convert to String and then to Boolean is going to evaluate to false or true.